superset-frontend/src/dashboard/components/Header.jsx

```diff
@@ -311,6 +311,7 @@ class Header extends React.PureComponent {
       }
 
       this.props.onSave(data, dashboardInfo.id, SAVE_TYPE_OVERWRITE);
+      window.location.reload();
```

Comment:
Unfortunately reload is not the right way to fix this issue:
The root cause is this:
- After the first save button click, dashboard metedata is updated in backend.
- But this updated metadata is not in redux store, so when you click 2nd save button, the original dashboard metadata is sent to backend, and overwrite the last update.

This fix is to add a page reload when 2nd save button is clicked. It just adds a race-condition: If page reload first, 2nd save request will be abandoned. But if 2nd save request comes first, metadata is still overwritten.  This fix will make issue more complicated.